File: wear/src/main/java/com/asamm/locus/addon/wear/features/trackRecord/recording/sensors/RecordingSensorStore.java
package com.asamm.locus.addon.wear.features.trackRecord.recording.sensors;

import android.content.Context;
import android.os.BatteryManager;

import com.asamm.locus.addon.wear.MainApplication;

import locus.api.utils.Logger;

import static android.content.Context.BATTERY_SERVICE;

/*
 * Created by milan on 03.08.2018.
 * This code is part of Locus project from Asamm Software, s. r. o.
 * Copyright (C) 2018
 */
public class RecordingSensorStore {
    public static volatile HrmDebugValue hrmDebug = new HrmDebugValue();
    public static volatile HrmValue hrm = new HrmValue();
    private static volatile BatteryValue battery = new BatteryValue();

    public static BatteryValue getBatteryValue() {
        if (!battery.isValid() || System.currentTimeMillis() - battery.timestamp > 30_000) {
            try {
                Context ctx = MainApplication.applicationContext;
                if (ctx != null) {
                    BatteryManager bm = (BatteryManager) ctx.getSystemService(BATTERY_SERVICE);
                    int batLevel = bm.getIntProperty(BatteryManager.BATTERY_PROPERTY_CAPACITY);
                    battery.setValue(batLevel);
                }

            } catch (Exception e) {
                Logger.INSTANCE.logE("RecordingSensorStore", "Battery level read failed", e);
                battery.setValue(BatteryValue.INVALID_VALUE);
            }
        }
        return battery;
    }
}

File: wear/src/main/java/com/asamm/locus/addon/wear/features/trackRecord/recording/sensors/RecordingSensorStore.kt
/*
 * Created by milan on 03.08.2018.
 * This code is part of Locus project from Asamm Software, s. r. o.
 * Copyright (C) 2018
 */
package com.asamm.locus.addon.wear.features.trackRecord.recording.sensors

import android.content.Context
import android.os.BatteryManager
import com.asamm.locus.addon.wear.MainApplication
import com.asamm.logger.Logger

object RecordingSensorStore {

    @JvmField
    @Volatile
    var hrmDebug = HrmDebugValue()

    @JvmField
    @Volatile
    var hrm = HrmValue()

    private val battery = BatteryValue()

    @JvmStatic
    val batteryValue: BatteryValue
        get() {
            if (!battery.isValid || System.currentTimeMillis() - battery.timestamp > 30000) {
                try {
                    val bm = MainApplication.app
                            .getSystemService(Context.BATTERY_SERVICE) as BatteryManager
                    val batLevel = bm.getIntProperty(BatteryManager.BATTERY_PROPERTY_CAPACITY)
                    battery.value = batLevel
                } catch (e: Exception) {
                    Logger.e(e, "RecordingSensorStore", "Battery level read failed")
                    battery.value = BatteryValue.INVALID_VALUE
                }
            }
            return battery
        }
}